testsuite/tests/typing-jkind-bounds/composite.ml

```diff
@@ -413,6 +414,8 @@ and 'a u = None | Some of 'a t
 |}]
 
 let foo (t : _ t @@ contended) = use_uncontended t
+(* CR reisenberg: This looks unsound, unless it's inferring a kind
+   of [immutable_data] for the argument but failing to print this.*)
```

Comment:
I think this is sound because a `'a t` doesn't ever actually contain a `'a`. I believe it's kind should just be `immutable_data` (although I am pleasantly surprised that this gets inferred correctly).